homeassistant/components/insteon/light.py

```diff
@@ -18,9 +20,20 @@
 
 async def async_setup_platform(hass, config, async_add_entities, discovery_info=None):
     """Set up the Insteon component."""
-    async_add_insteon_entities(
-        hass, DOMAIN, InsteonDimmerEntity, async_add_entities, discovery_info
-    )
+
+
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Set up the Insteon lights from a config entry."""
+
+    def add_entities(discovery_info=None):
+        """Add the Insteon entities for the platform."""
+        async_add_insteon_entities(
+            hass, DOMAIN, InsteonDimmerEntity, async_add_entities, discovery_info
+        )
+
+    signal = f"{SIGNAL_ADD_ENTITIES}_{DOMAIN}"
+    async_dispatcher_connect(hass, signal, add_entities)
+    add_entities()
```

Comment:
Is there a reason you can't call `async_add_insteon_entities` directly here?

Reply:
The underlying protocol can add devices (and therefore entities) dynamically. The dispatcher is used to allow HA to see a new device is created and add the appropriate entities. I need the handle to `async_add_entities` above to make that happen. This is what balloob recommended when I asked in the discord channel. The only other way I can think to do it is to keep a link to each platform's `async_add_entities` in hass.data but this feels cleaner.

Reply:
Makes sense 👍